book/src/qobject/cxxqtthread.md

```diff
@@ -7,15 +7,23 @@ SPDX-License-Identifier: MIT OR Apache-2.0
 
 # CxxQtThread
 
-`CxxQtThread<T>` is used for [threading](../concepts/threading.md) with QObject#s and allows you to queue events from a background thread to occur on the Qt event loop.
+`CxxQtThread<T>` is used for easy [threading](../concepts/threading.md) with QObjects.
+The QObjects generated by CXX-Qt are neither [`Sync`](https://doc.rust-lang.org/std/marker/trait.Sync.html) nor [`Send`](https://doc.rust-lang.org/std/marker/trait.Send.html).
+Therefore they may not be passed between threads.
 
-To access the `CxxQtThread<T>` use the `qt_thread(&self)` method on the [`CppObj`](./cpp_object.md) or on the C++ pointer of the QObject.
+`CxxQtThread<T>` is a handle to the Qt thread that the `qobject::T` lives on.
+This handle is `Send` and can therefore be moved into other threads.
+It allows you to queue events from a different thread to occur on the thread of the `qobject::T` by using the Qt Event Loop.
+
+To access the `CxxQtThread<T>` use the `qt_thread(&self)` method on a [`qobject::T`](./cpp_object.md).
 
 ```rust,ignore,noplayground
 {{#include ../../../examples/qml_features/rust/src/threading.rs:book_qt_thread}}
 ```
 
-The `CxxQtThread<T>` can then be moved into the Rust thread, a `queue(fn(ctx: Pin<&mut TQt>) -> bool` is used to queue a Rust function pointer onto the Qt event loop. The first argument in the function pointer is a pinned pointer to the C++ side of the QObject.
+The `CxxQtThread<T>` can then be moved into any Rust thread.
+The `queue(impl FnOnce(ctx: Pin<&mut TQt>)) -> bool` is used to queue a Rust function pointer onto the Qt event loop. The first argument in the function pointer is a pinned pointer to the C++ side of the QObject.
```

Comment:
What is the meaning of the returned bool?

Reply:
Updated, it actually returns a Result now.
Basically, whether it was possible to queue the lambda or not.
Most of the time a failure means the QObject doesn't exist anymore.